src/ScheduledTasks/Client/ScheduledTaskClient.cs

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.DurableTask.Client;
+using Microsoft.DurableTask.Client.Entities;
+using Microsoft.DurableTask.Entities;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DurableTask.ScheduledTasks;
+
+/// <summary>
+/// Client for managing scheduled tasks in a Durable Task application.
+/// </summary>
+public class ScheduledTaskClient : IScheduledTaskClient
+{
+    readonly DurableTaskClient durableTaskClient;
+    readonly ILogger logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ScheduledTaskClient"/> class.
+    /// </summary>
+    /// <param name="durableTaskClient">The Durable Task client to use for orchestration operations.</param>
+    /// <param name="logger">logger.</param>
+    public ScheduledTaskClient(DurableTaskClient durableTaskClient, ILogger logger)
+    {
+        this.durableTaskClient = Check.NotNull(durableTaskClient, nameof(durableTaskClient));
+        this.logger = Check.NotNull(logger, nameof(logger));
+    }
+
+    /// <inheritdoc/>
+    public async Task<ScheduleHandle> CreateScheduleAsync(ScheduleCreationOptions creationOptions, CancellationToken cancellation = default)
+    {
+        Check.NotNull(creationOptions, nameof(creationOptions));
+        this.logger.ClientCreatingSchedule(creationOptions);
+
+        try
+        {
+            string scheduleId = creationOptions.ScheduleId;
+            EntityInstanceId entityId = new EntityInstanceId(nameof(Schedule), scheduleId);
+
+            // Check if schedule already exists
+            bool scheduleExists = await this.CheckScheduleExists(scheduleId, cancellation);
+            if (scheduleExists)
+            {
+                throw new ScheduleAlreadyExistException(scheduleId);
+            }
+
+            // Call the orchestrator to create the schedule
+            ScheduleOperationRequest request = new ScheduleOperationRequest(entityId, nameof(Schedule.CreateSchedule), creationOptions);
+            string instanceId = await this.durableTaskClient.ScheduleNewOrchestrationInstanceAsync(
```

Comment:
I was expecting schedules to be implemented using simple entities, but I see you're creating an orchestration. Why is the orchestration needed?

Reply:
It serves as an indirect method for achieving a poll-free signal operation call, as recommended by @sebastianburckhardt. Within the orchestration entity context, it can utilize the CallEntity API, which not only eliminates polling for the consumer but also manages tracking, retrieval, and decoupling of feedback. This ensures that the operation status, including the result or failure details, is captured at the entity operation level, rather than being embedded within the entity state, as was done in my old polling-based approach.